jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -963,6 +971,21 @@ private static final MethodHandle resolveInvokeDynamic(Class<?> clazz, String na
 		return result;
 	}
 	
+	/*
+	 * Throw NoClassDefFoundError if the cause of TypeNotPresentException is 
+	 * ClassNotFoundException. Otherwise, re-throw TypeNotPresentException.
+	 */
+	private static void throwNoClassDefFoundError(TypeNotPresentException e) {
+		Throwable cause = e.getCause();
+		if (cause instanceof ClassNotFoundException) {
+			NoClassDefFoundError noClassDefFoundError = new NoClassDefFoundError(cause.getMessage());
+			noClassDefFoundError.initCause(cause);
+			throw noClassDefFoundError;
+		} else {
```

Comment:
Since the `if` block unconditionally throws, the `else` isn't necessary.  The `throw e` can follow the end of the if block.  This makes again makes it clearer that there is no fall through path here.

Reply:
fixed.